src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -341,10 +341,13 @@ private BoundStatement RewriteUsingStatementTryFinally(SyntaxNode syntax, BoundB
                     hasErrors: false);
             }
 
+            // Terminate the try block with a hidden sequence point
+            BoundBlock terminatedTryBlock = tryBlock.WithStatements(tryBlock.Statements.Add(new BoundSequencePoint(null, null)));
```

Comment:
I'm able to repro the problem in the debugger for this scenario:
```C#
        void F3(bool x)
        {
            string y = "";
            lock (y)
                if (!x)
                    System.Console.Write(1);
                else
                    System.Console.Write(2);             
        }
```
(lowering lock introduces a try/catch)